specs/vm/instruction_set.md

```diff
@@ -702,6 +705,20 @@ Otherwise, advance the program counter `$pc` by `4`.
 
 See also: [BIP-65](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki) and [Bitcoin's Time Locks](https://prestwi.ch/bitcoin-time-locks).
 
+### J: Jump
+
+|             |                                                     |
+|-------------|-----------------------------------------------------|
+| Description | Jumps to the code instruction offset by a register. |
+| Operation   | ```$pc = $is + $rA * 4;```                          |
```

Comment:
Should we concern with the case of `$rA == 0` that will cause an infinite loop and exhaust the gas or should we treat that as a malformed runtime execution? Note that this can't be trivially spotted at compilation time since it depends on runtime values.

Before, it was trivial to catch these cases since the jumps were immediate; therefore, part of the hardcoded program.

Reply:
It's not an issue since it's an application logic error, and DoS will be prevented by gas metering execution.

Reply:
Note also that the intended use of these instructions by a sane compiler is:
1. Load function pointer into register. Load current pc into a return register. Then jump to function pointer register.
2. When returning, jump to return register.

So the set of possible memory addresses is known at compile time.

Reply:
The actual `call` part will remain static most of the time, it's more about `ret`.  So (1) will most of the time just be 'save `pc` somewhere, jump to function'.